hudi-common/src/main/java/org/apache/hudi/common/util/HoodieWriteClientUtils.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.util;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.hudi.common.model.HoodieCommitMetadata;
+import org.apache.hudi.common.table.HoodieTimeline;
+import org.apache.hudi.common.table.timeline.HoodieInstant;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+
+
+/**
+ * Helper class for utilities required during write time
+ */
+public class HoodieWriteClientUtils {
+
+  private static final Logger LOG = LogManager.getLogger(HoodieWriteClientUtils.class);
+
+  /**
+   * Obtains the average record size based on records written during previous commits. Used for
+   * estimating how many records pack into one file.
+   */
+  public static long averageBytesPerRecord(HoodieTimeline commitTimeline, int defaultRecordSizeEstimate) {
+    long avgSize = 0L;
+    try {
+      if (!commitTimeline.empty()) {
+        // Go over the commits in reverse order to get a more recent estimate of average record size.
+        List<HoodieInstant> instants = commitTimeline.getInstants().collect(Collectors.toList());
+        Collections.reverse(instants);
+        for (HoodieInstant instant : instants) {
+          HoodieCommitMetadata commitMetadata = HoodieCommitMetadata
+              .fromBytes(commitTimeline.getInstantDetails(instant).get(), HoodieCommitMetadata.class);
+          long totalBytesWritten = commitMetadata.fetchTotalBytesWritten();
+          long totalRecordsWritten = commitMetadata.fetchTotalRecordsWritten();
+          if (totalBytesWritten > 0 && totalRecordsWritten > 0) {
+            avgSize = (long) Math.ceil((1.0 * totalBytesWritten) / totalRecordsWritten);
+            break;
+          }
+        }
+      }
+    } catch (Throwable t) {
+      // make this fail safe.
+      LOG.error("Error trying to compute average bytes/record ", t);
+    }
+    return avgSize <= 0L ? defaultRecordSizeEstimate : avgSize;
```

Comment:
instead may be just initialize avgSize to defaultRecordSizeEstimate ?  and return avgSize here?

Reply:
yes. Thats neat.